src/com/jdngray77/htmldesigner/backend/BackgroundTask.kt

```diff
@@ -15,23 +22,52 @@ object BackgroundTask : Subscriber {
     }
 
 
+    /**
+     * ### Executes a [runnable] task.
+     *
+     * Executes the given task on a new thread, a pooled thread or the calling thread, at the discretion of the
+     * [threadPool]. There are no guarantees that the task will be executed immediately.
+     */
+    fun execute(runnable: Runnable) {
+        threadPool.execute(runnable)
+    }
+
+
+    @Deprecated("Currently not implemented.", ReplaceWith("TODO()"))
     fun runOnUIThread(runnable: Runnable) {
         TODO()
     }
 
-    
+
     /**
      * Submits a [runnable] task to the for [threadPool] for execution.
+     *
+     * If the thread pool has been instructed to or is already shutdown, the task will not be submitted.
      */
+    @Synchronized
     fun submit(runnable: Runnable) {
-        threadPool.submit(runnable)
+        if (!threadPool.isShutdown)
+            threadPool.submit(runnable)
+    }
+
+
+    /**
+     * Returns all tasks waiting for execution on a worker [Thread] in the [threadPool].
+     *
+     * @return Tasks waiting in the [threadPool] queue for execution.
+     */
+    fun scheduledTasks(): BlockingQueue<Runnable> {
+        val scheduled = threadPool as ThreadPoolExecutor
```

Comment:
Im gonna labmda this up so it doesn't waste time adding creating a variable on the stack that isn't really needed